src/ClientOptionsTrait.php

```diff
@@ -317,4 +351,25 @@ private function isBackwardsCompatibilityMode(): bool
     {
         return false;
     }
+
+    /**
+     * @param null|false|LoggerInterface $logger
+     * @param string $options
+     */
+    private function logConfiguration(null|false|LoggerInterface $logger, string $options): void
+    {
+        if (!$logger) {
+            return;
+        }
+
+        $configurationLog = [
+            'timestamp' => date(DATE_RFC3339),
+            'severity' => strtoupper(LogLevel::DEBUG),
+            'jsonPayload' => [
+                'suppliedConfiguration' => $options
+            ]
```

Comment:
Let's add `processId` here as well, so it can be at least loosely tied to the request/response logs